front50-core/src/main/java/com/netflix/spinnaker/front50/validator/AppNameValidator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.front50.validator;
+
+import com.google.common.base.Strings;
+import com.netflix.spinnaker.front50.model.application.Application;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AppNameValidator implements ApplicationValidator {
+  private String validationRegEx;
+
+  @Autowired
+  public AppNameValidator(
+      @Value("${validation.app-name-validator.name-reg-ex:#{null}}}") String validationRegEx) {
+    this.validationRegEx = validationRegEx;
+  }
+
+  @Override
+  public void validate(Application application, ApplicationValidationErrors validationErrors) {
+    if (Strings.isNullOrEmpty(validationRegEx)) {
+      return;
+    }
+
+    String appName = Optional.ofNullable(application.getName()).orElse("");
+    if (!appName.matches(validationRegEx)) {
+      validationErrors.rejectValue(
+          "name",
+          "application.name.invalid",
+          "Application name can only contain a-z, 0-9, dots, and dashes");
```

Comment:
should this message reference the validation regex instead ?

Reply:
doh, thanks! i changed to regex and then didn't update the message